cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java

```diff
@@ -136,16 +138,34 @@ public void stop(ProgramRunInfo programRunInfo) throws Exception {
     publishToControlChannel(message);
   }
 
+  /**
+   * This method ends up actually killing the process because it uses RuntimeJobStatus.RUNNING
+   * */
   @Override
   public void kill(ProgramRunInfo programRunInfo) throws Exception {
-    RuntimeJobDetail jobDetail = getDetail(programRunInfo).orElse(null);
+    RuntimeJobDetail jobDetail = new RuntimeJobDetail(programRunInfo, RuntimeJobStatus.RUNNING);
```

Comment:
Why RUNNING?

Reply:
this is to make sure we cover the case where we actually want to kill() the program and not just clean(). 

To be honest, I feel the flow is more confusing in this case. In the previous PR where we had separately named functions like clean() and kill(), the intention as to what we were doing was clearer to someone reading the code the first time. 

Now in effect we are doing the same thing, but with convoluted logics. 

The main reason we have to do this is because getDetail() does not give us a real state of affairs. That is true for both the earlier PR and this one. 


We are now distinguishing between kill() and clean() by doing 
kill(programRunInfo) and kill(runTimeJobDetail) instead.